### DISCUSSION TITLE
How to update the text?

### DISCUSSION THREAD

**Discussion by Theodliu** [Kitware/trame#344] [Trame/VTK] [answered]
Hi I am new to trame. I refered to this : https://github.com/Kitware/trame/discussions/303 but it doesn't work for me.
Here's my code:
`class MouseInteractorHighLightActor(vtkInteractorStyleTrackballCamera):

    def __init__(self, parent=None):
        self.AddObserver("LeftButtonPressEvent", self.onLeftButtonDown)
        self.LastPickedActor = None

    def onLeftButtonDown(self, obj, event):
        clickPos = self.GetInteractor().GetEventPosition()

        #  Pick from this location.
        picker = vtkPropPicker()
        picker.Pick(clickPos[0], clickPos[1], 0, self.GetDefaultRenderer())
        self.NewPickedActor = picker.GetActor()
        if (self.NewPickedActor != None):
            self.LastPickedActor = self.NewPickedActor
        self.OnLeftButtonDown()
        state.current_pickername = self.NewPickedActor.name
        print(state.current_pickername) #<--it prints the correct name
        ctrl.view_update()
        return

server = get_server()
state.current_pickername = "undefined"

with SinglePageLayout(server) as layout:
  with layout.content:
  ...
  layout.footer.set_text("The dynamic text will be next: {{ current_pickername }}")`

The text I got is: The dynamic text will be next: undefined but it won't update anymore.
Thank you for your help.

Edited:
Now I find that if I add a button.
with vuetify.VBtn(
                        icon=True, click=clear_all):  ##here clear_all can be any function
                        vuetify.VIcon("mdi-undo-variant")
Everytime I click the button, the text will update. However, the text can't update itself once its value changed.

**Comment by jourdain** [comment]
```python
with state:
   state.current_pickername = self.NewPickedActor.name
```

**Reply by Theodliu** [reply]
It works, thanks a lot.

**Reply by Theodliu** [reply]
Sorry but I have a new question here. I am currently working on the right-click menu. My idea is that when I right click, I will get the position of the mouse and update the position with the help of state.menupositionx and state.menupositiony
Here's my code:

`            with html.Div(style="position: absolute;top: 400px; left: {{menupositionx}} px;width: 400px;background:white;max-height:80vh"):
                html.Template(
                    "<div style='padding-left:0px;padding-top: {{menupositiony}}px ;font-size:20px;color:rgba(0,0,0,.87);padding-bottom:16px'>{{menupositionx}}</div>")`

Now, the state enables me to update the text but it seems that it won't move any more.

Thanks a lot again.

**Reply by jourdain** [reply, marked as answer]
the `... {{ }} ...` are only for text expansion into the UI. If you want a javascript expression as an attribute in vue, you must use the `v-bind:propName` or `:propName`, which you get with the tuple notation in trame.

You can look at [that example](https://kitware.github.io/trame/examples/vtk/picking.html), which uses a flying tooltip that follows the mouse.

**Reply by Theodliu** [reply]
Solved. Thanks so much.
